### PR TITLE
Add support for DF icons with `text` (initials) without breaking `Select` `Option`

### PR DESCRIPTION
**Pull request by JulianG-TW**
## Context

The DFv3 specification for **Icons** states that the icon object may contain a `name` or a `text` property. Until now we only supported `name`, which rendered an icon.

## Changes

We now support `text` and render an `Avatar` component of type `initials`.

In addition we also render icons wrapped in an `Avatar` component, this time, of type `icon`. This is to align with the [Neptune guidelines](https://neptune.wise.com/components/navigation-option/). See also [Decision](https://transferwise.github.io/neptune-web/components/navigation/Decision#Decision) and [Avatar](https://transferwise.github.io/neptune-web/components/content/Avatar#Avatar).

### Before

before

### After

after

**Text icons were not supported in "radio" buttons nor in dropdowns**

Screenshot 2022-06-24 at 11 10 43

## Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [x] Changes work in all supported browsers (don't forget IE11)
- [x] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/jsonSchemaForm/schemaFormControl/optionMapper.js [hunk 1]
```diff
@@ -61,6 +62,11 @@ export const mapIconToAvatar = (icon) => {
       ),
     };
   }
+  if (icon?.text) {
+    return {
+      avatar: <Avatar type={AvatarType.INITIALS}>{icon.text}</Avatar>,
```
Comment (fergusjordantw) [inline packages/dynamic-flows/src/jsonSchemaForm/schemaFormControl/optionMapper.js:67]: Will the text always fit in an Avatar as one or two characters? Or should we be splitting the string into initials for safety?

Comment (fergusjordantw) [reply packages/dynamic-flows/src/jsonSchemaForm/schemaFormControl/optionMapper.js:67]: I guess it's fine to leave up to whoever is passing down the schema

File: packages/dynamic-flows/src/layout/decision/index.js [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+        <DynamicIcon type={option.icon.name} />
+      </Avatar>
+    );
+  }
+  if (option.icon?.text) {
+    return <Avatar type={AvatarType.INITIALS}>{option.icon.text}</Avatar>;
+  }
+  if (option.image?.url) {
     // eslint-disable-next-line jsx-a11y/img-redundant-alt
     return <img src={option.image?.url} alt={option.image?.text} />;
```
Comment (fergusjordantw) [inline packages/dynamic-flows/src/layout/decision/index.js:26]: Out of scope of this change but should this also be thumbnail avatar? Any reason it isn't?

**Review by fergusjordantw — APPROVED** [review]
Didn't look at the schema changes in detail

File: packages/dynamic-flows/src/jsonSchemaForm/schemaFormControl/optionMapper.js (hunk 1 above)
Comment (JulianG-TW) [reply packages/dynamic-flows/src/jsonSchemaForm/schemaFormControl/optionMapper.js:67]: Exactly. Neptune Web Avatar doesn't care if the text fits either.
It's up to the service generating the schema.
image

File: packages/dynamic-flows/src/layout/decision/index.js (hunk 2 above)
Comment (JulianG-TW) [reply packages/dynamic-flows/src/layout/decision/index.js:26]: Definitely out of scope. We could use Avatar here as well:
image

But I need to check what mobile clients are doing. I've seen this used for brands/logos, and I'm not sure we can assume the can be masked into a circle.

**Comment by JulianG-TW** [comment]
Thanks for reviewing @fergusjordantw !

**Review by AllanHasegawaTw — APPROVED** [review]

**Pull request merged**